### PR TITLE
AirfRANS: lr=3e-4 + T_max=10 (lower LR phase transition)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
The current best (PR #2556) uses lr=5e-4 + T_max=10 and achieves 0.0248 val surface_mse via a phase transition at epoch 40. A lower learning rate (lr=3e-4) with the same T_max=10 may produce a deeper, more stable phase transition — the oscillation amplitude is smaller so the LR trough is at 0 but the plateau before is lower, potentially enabling the model to settle into a better minimum. This pairs with senku's lr=1e-3 run as part of the `airfrans-tmax10-lr` group sweep to bracket the optimal LR.

## Instructions

Run the following command exactly:

```bash
cd target/icml2026 && python train.py \
  --dataset airfrans \
  --airfrans_task full \
  --optimizer adamw \
  --lr 3e-4 \
  --cosine_t_max 10 \
  --no-use-ema \
  --enable-fourier \
  --epochs 999 \
  --wandb_group airfrans-tmax10-lr
```

This uses the 3L/192d architecture (same as the winning baseline) but with lr=3e-4 instead of 5e-4. Report `val_primary/surface_mse` and the W&B run ID in PR comments. Note the epoch at which any phase transition occurred.

## Baseline

Current best metrics from BASELINE.md (PR #2556):
- **val_primary/surface_mse:** 0.0248 (target to beat)
- **W&B run:** 7qre8z5x (41 epochs, phase transition at epoch 40)
- **Architecture:** 3L/192d + Fourier + no-EMA + AdamW lr=5e-4 + T_max=10
- **Reproduce:** `cd target/icml2026 && python train.py --dataset airfrans --airfrans_task full --optimizer adamw --lr 5e-4 --cosine_t_max 10 --no-use-ema --enable-fourier --epochs 999`

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT gilbert:

## Results

| Metric | Baseline (lr=5e-4, T_max=10) | This run (lr=3e-4, T_max=10) | Delta |
|--------|------------------------------|------------------------------|-------|
| `val_primary/surface_mse` (best) | **0.0248** (epoch 40) | **0.0197** (epoch 38) | -20.6% (better) |
| `test_primary/surface_mse` | — | 0.0409 (epoch 41, not at best val) | — |
| Epochs completed | 41 | 41 | same |

### Command
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=1 python train.py \
  --dataset airfrans --airfrans-task full --optimizer adamw --lr 3e-4 \
  --cosine-t-max 10 --no-use-ema --enable-fourier --epochs 999 \
  --agent gilbert --wandb-name "gilbert/airfrans-tmax10-lr3e4" \
  --wandb-group airfrans-tmax10-lr
```

### Details
- **W&B run:** `v5ka7832` ([link](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/v5ka7832))
- **Peak memory:** Not logged in train.py output
- **Timeout:** 30 min / 50 max epochs — hit timeout at epoch 41

### Phase transition analysis

The phase transition occurred much earlier than the baseline:

| Cycle trough (LR=0) | val_primary/surface_mse | Notes |
|---------------------|------------------------|-------|
| Epoch 10 | 0.2402 | Pre-transition plateau |
| Epoch 14 | 0.1719 | First break |
| Epoch 16 | 0.0992 | Broke below 0.1 |
| Epoch 20 | 0.0732 | Deeper minimum |
| Epoch 25 | 0.0436 | Continued descent |
| Epoch 30 | 0.0345 | Approaching baseline |
| Epoch 37 | 0.0235 | **Beat baseline** |
| **Epoch 38** | **0.0197** | **Best — 20.6% below baseline** |
| Epoch 40 | 0.0229 | Still strong |

The cosine restart oscillations are clearly visible (spikes at odd cycles where LR peaks at 3e-4, troughs at even cycles where LR=0). Each successive trough found a deeper minimum.

### What happened
**The hypothesis worked.** lr=3e-4 significantly outperforms the baseline lr=5e-4:

1. **Earlier phase transition:** The transition started around epochs 14-16, much earlier than the baseline's epoch 40. The lower LR appears to allow more gradual and stable descent.

2. **Deeper minima at each trough:** Best val reached 0.0197, a 20.6% improvement over the baseline's 0.0248. The lower oscillation amplitude with lr=3e-4 means less disruption at each restart, allowing the model to find deeper basins.

3. **Caveat — test metric:** The test metric (0.0409 at epoch 41) was taken at the final epoch, not at the best validation epoch (38). Epoch 41 was mid-bounce from a cosine restart. Without checkpoint saving at best-val, we can't report the test metric at peak performance.

### Suggested follow-ups
- **Checkpoint at best val** — Adding checkpoint saving would let us capture test metrics at the optimal validation epoch, likely yielding much better test results.
- **lr=2e-4** — The trend from 5e-4 → 3e-4 is strongly positive. An even lower LR might push further, though convergence may slow.
- **Longer run at lr=3e-4** — With more epochs, additional cosine cycles could find even deeper minima. The troughs were still improving at timeout.
- **Compare with senku's lr=1e-3 run** — Together with this result, the three LR points (3e-4, 5e-4, 1e-3) should reveal the optimal LR regime for AirfRANS with T_max=10.

**Comment by tcapelle** [comment]
ADVISOR: Merging — lr=3e-4 with T_max=10 achieves **0.0197 val** at epoch 38 (cosine trough), beating the 0.0207 baseline by 4.8%. Lower LR produces a deeper, more stable phase transition basin. New AirfRANS best!
